### DISCUSSION TITLE
Ability to change the color of an entire column

### DISCUSSION THREAD

**Discussion by unoffending** [opentoonz/opentoonz#6459] [Ideas]
### Is your feature request related to a problem? Please describe

Sometimes it is very easy to lose a certain column, or columns may have the same purpose (for example, background or line).

### Describe the solution you'd like

Make it possible to change the color of an entire column to make it easier to navigate. I took this feature from After Effects and it seems to me that it can be useful in OpenToonz, especially when there are a lot of layers.

**Comment by RodneyBaker** [comment]
This is a feature I use a lot in Opentoonz.

We do this via column color (and transparency).
I'm attaching a project file that has just one level spread across columns and frames that has a black circle in each frame.
Using color tinting (and transparency) via the the dropdown menu at top of xsheet column I've changed the color of each column.

A few additional notes/considerations:
Column Color Tint and Transparency will only be seen on canvas and not in Prview and Final Render UNLESS we turn that option on in Scene Settings.  Scene Settings is also where we can and (and change) the colors available in the dropdown.

So... drawing only in solid black with column color and transparency we get things like:  

![ColumnColorTintTransparency](https://github.com/opentoonz/opentoonz/assets/2576432/6d04d997-5977-476b-891d-d3f27ad1bc1d)

[ColumnColorTintingTransparency.zip](https://github.com/opentoonz/opentoonz/files/15191292/ColumnColorTintingTransparency.zip)

![image](https://github.com/opentoonz/opentoonz/assets/2576432/4f668ab7-8e3a-4a79-a208-86ac8ea24a43)

**Comment by unoffending** [comment]
> This is a feature I use a lot in Opentoonz.
> 
> We do this via column color (and transparency). I'm attaching a project file that has just one level spread across columns and frames that has a black circle in each frame. Using color tinting (and transparency) via the the dropdown menu at top of xsheet column I've changed the color of each column.
> 
> A few additional notes/considerations: Column Color Tint and Transparency will only be seen on canvas and not in Prview and Final Render UNLESS we turn that option on in Scene Settings. Scene Settings is also where we can and (and change) the colors available in the dropdown.
> 
> So... drawing only in solid black with column color and transparency we get things like:
> 
> ![ColumnColorTintTransparency](https://private-user-images.githubusercontent.com/2576432/327510674-6d04d997-5977-476b-891d-d3f27ad1bc1d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2Njg5OTcsIm5iZiI6MTcxNDY2ODY5NywicGF0aCI6Ii8yNTc2NDMyLzMyNzUxMDY3NC02ZDA0ZDk5Ny01OTc3LTQ3NmItODkxZC1kM2YyN2FkMWJjMWQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDJUMTY1MTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGI4M2U0YzAzYzgwZjQ1NDc0MzQ3ZjJhMzYxM2MzZTc4MDRlMjgyNjI4Zjk5ZDlkZWM0NTdiNWMzODMyN2Q5YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oNZSJ6lS862H9oGrUnRTqd4MY9sb5cZulgjena3e_Y4) [ ![ColumnColorTintTransparency](https://private-user-images.githubusercontent.com/2576432/327510674-6d04d997-5977-476b-891d-d3f27ad1bc1d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2Njg5OTcsIm5iZiI6MTcxNDY2ODY5NywicGF0aCI6Ii8yNTc2NDMyLzMyNzUxMDY3NC02ZDA0ZDk5Ny01OTc3LTQ3NmItODkxZC1kM2YyN2FkMWJjMWQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDJUMTY1MTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGI4M2U0YzAzYzgwZjQ1NDc0MzQ3ZjJhMzYxM2MzZTc4MDRlMjgyNjI4Zjk5ZDlkZWM0NTdiNWMzODMyN2Q5YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oNZSJ6lS862H9oGrUnRTqd4MY9sb5cZulgjena3e_Y4) ](https://private-user-images.githubusercontent.com/2576432/327510674-6d04d997-5977-476b-891d-d3f27ad1bc1d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2Njg5OTcsIm5iZiI6MTcxNDY2ODY5NywicGF0aCI6Ii8yNTc2NDMyLzMyNzUxMDY3NC02ZDA0ZDk5Ny01OTc3LTQ3NmItODkxZC1kM2YyN2FkMWJjMWQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDJUMTY1MTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGI4M2U0YzAzYzgwZjQ1NDc0MzQ3ZjJhMzYxM2MzZTc4MDRlMjgyNjI4Zjk5ZDlkZWM0NTdiNWMzODMyN2Q5YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oNZSJ6lS862H9oGrUnRTqd4MY9sb5cZulgjena3e_Y4) [ ](https://private-user-images.githubusercontent.com/2576432/327510674-6d04d997-5977-476b-891d-d3f27ad1bc1d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2Njg5OTcsIm5iZiI6MTcxNDY2ODY5NywicGF0aCI6Ii8yNTc2NDMyLzMyNzUxMDY3NC02ZDA0ZDk5Ny01OTc3LTQ3NmItODkxZC1kM2YyN2FkMWJjMWQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDJUMTY1MTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGI4M2U0YzAzYzgwZjQ1NDc0MzQ3ZjJhMzYxM2MzZTc4MDRlMjgyNjI4Zjk5ZDlkZWM0NTdiNWMzODMyN2Q5YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oNZSJ6lS862H9oGrUnRTqd4MY9sb5cZulgjena3e_Y4)
> 
> [ColumnColorTintingTransparency.zip](https://github.com/opentoonz/opentoonz/files/15191292/ColumnColorTintingTransparency.zip)
> 
> ![image](https://private-user-images.githubusercontent.com/2576432/327512294-4f668ab7-8e3a-4a79-a208-86ac8ea24a43.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2Njg5OTcsIm5iZiI6MTcxNDY2ODY5NywicGF0aCI6Ii8yNTc2NDMyLzMyNzUxMjI5NC00ZjY2OGFiNy04ZTNhLTRhNzktYTIwOC04NmFjOGVhMjRhNDMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDJUMTY1MTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTI5OTkyMTE2YzcxMWYxYzRiNmY4ZTQ0NGI4MjM1NzJlZTczOTk2NmI1MjdlNTA3ZGQ5Y2UxNWRhN2U5OGFlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.OI4CvOORNR5M2CuRJDOqC44rswr9qKZIuJlyJbzgC50)

It's a nice feature and I like it! But that's not what I meant...
I meant to change the color of the columns themselves... As we know, the cells of the Toonz Raster level are green, the Raster level are blue, and the Vector level are yellow... But what if it was possible to change their color?... Then, this could help navigate the columns, especially if there are a lot of them...

**Comment by unoffending** [comment]
@RodneyBaker 
![after effects](https://github.com/opentoonz/opentoonz/assets/109032978/0104180a-1f75-4aa5-adb0-585803f4dde0)
For example, in this frame.... Some columns are yellow.... They became like this because they were so marked...

**Comment by RodneyBaker** [comment]
For now I guess we'll have to live with tagging the columns with that little color swatch at the top of the column.

It's a good idea to be able to customize/color everything we can so perhaps someone will seize the opportunty and develop the idea. I would think we'd want some strong controls and filters to manage the number of columns that can be set to different colors as I can see that quickly getting out of hand.  One of the reasons levels are colored the way they are now is to distinguish what the column contains.  Your feature would apparently negate that in favor of overriding that color.

This might be something that is currently being implemented in Tahoma2D to some extent through Column Folders.
I'm not aware those folders can be colored individually however.

**Comment by unoffending** [comment]
> For now I guess we'll have to live with tagging the columns with that little color swatch at the top of the column.
> 
> It's a good idea to be able to customize/color everything we can so perhaps someone will seize the opportunty and develop the idea. I would think we'd want some strong controls and filters to manage the number of columns that can be set to different colors as I can see that quickly getting out of hand. One of the reasons levels are colored the way they are now is to distinguish what the column contains. Your feature would apparently negate that in favor of overriding that color.
> 
> This might be something that is currently being implemented in Tahoma2D to some extent through Column Folders. I'm not aware those folders can be colored individually however.

Thank you! But I think that then such a function can be replaced with folders...

**Comment by RodneyBaker** [comment]
Slightly unrelated aside:  We can change the color of levels vis CSS in Preferences > Additional Style Sheets
This won't change the color of everything in the column however... only the levels in the column.

**Comment by RodneyBaker** [comment]
> Thank you! But I think that then such a function can be replaced with folders...

It will take awhile for Folders to get fully developed and even more time for it to be ported from T2D to OT.
@manongjohn rightly prefers to have new features he's created live in T2D for at least a little while before being ported.
This is a good thing because it allows time to test, verfiy... even further enhance... features as they are ported.

If you wish to test the folder feature many Opentoonz users use both Opentoonz and Tahoma2D simultaneously.
It may not yet be available in a regular release but will be available in some capacity (perhaps a nightly release)
I believe the Folders addition is scheduled for the T2D v1.5 release in December.

**Comment by unoffending** [comment]
> > Thank you! But I think that then such a function can be replaced with folders...
> 
> It will take awhile for Folders to get fully developed and even more time for it to be ported from T2D to OT. @manongjohn rightly prefers to have new features he's created live in T2D for at least a little while before being ported. This is a good thing because it allows time to test, verfiy... even further enhance... features as they are ported.
> 
> If you wish to test the folder feature many Opentoonz users use both Opentoonz and Tahoma2D simultaneously. It may not yet be available in a regular release but will be available in some capacity (perhaps a nightly release) I believe the Folders addition is scheduled for the T2D v1.5 release in December.

Thank you!

**Comment by unoffending** [comment]
> Hmm, as I see it OT basically has most of the heavy lifting in place. It can already tint columns to differentiate their types, and it already has a dropdown for tinting the contents of the column.
> 
> With a little bit of glue and a toggle, this could be wrapped into either a global option, or a column option, with the global option being the easier to implement.
> 
> And while I can't think of a snappy snazzy name for that option right now, it would just be a checkbox that, when ticked, does not apply color to the content of the column, but the column itself.
> 
> Perhaps this can be considered as a 'hacky' fallback of these 'folders' that were mentioned never make it into OT. I assume its like layers, and layer groups and you put/drag columns into those folders to group them. This, to me smells like it's more geared toward the timeline (I'm a xsheet kinda guy).
> 
> Ah, yes, found it [tahoma2d/tahoma2d#1385](https://github.com/tahoma2d/tahoma2d/pull/1385)
> 
> Well...I think this is incredibly ugly in both implementations, even the 'better' timeline one (which again I assumed was the first thought)
> 
> I guess if that were the way to go then I would not want folders at all and would prefer just a column tint at most.
> 
> But to fully honest, this, like many issues, is more of a diligence thing. Sub xsheets already exist... they take a little bit more work but, in my opinion, look less like salad than that, and I'm not even sure that my grognarditute is the majority in my opinion here. It really does look like salad/noise.
> 
> Hmmmmmmmmmmmmmmmst.

Thank you! I downloaded the nightly build
![folder](https://github.com/opentoonz/opentoonz/assets/109032978/bb6084c8-f5bc-4f67-87aa-269692828a76)

**Comment by ForestEco** [comment]
It may be useful to have the ability to sort the priority of colors, so if multiple columns share color priority, then the FX could be batched, so you don't need to make multiple nodes per column. That way some FX could be updated.
